server/src/main/java/com/cloud/storage/StorageManagerImpl.java

```diff
@@ -648,6 +651,38 @@ public boolean stop() {
         return true;
     }
 
+    private DataStore createLocalStorage(Map<String, Object> poolInfos) throws ConnectionException{
+        Object existingUuid = poolInfos.get("uuid");
+        if( existingUuid == null ){
+            poolInfos.put("uuid", UUID.randomUUID().toString());
+        }
+        String hostAddress = poolInfos.get("host").toString();
+        Host host = _hostDao.findByName(hostAddress);
+
+        if( host == null )
+            host = _hostDao.findByIp(hostAddress);
+
+        if( host == null )
+            host = _hostDao.findByPublicIp(hostAddress);
+
+        if( host == null )
+            throw new InvalidParameterValueException(String.format("host %s not found",hostAddress));
```

Comment:
```suggestion
        if( host == null ) {
            host = _hostDao.findByIp(hostAddress);

          if( host == null ) {
              host = _hostDao.findByPublicIp(hostAddress);

            if( host == null ) {
                throw new InvalidParameterValueException(String.format("host %s not found",hostAddress));
                }
              }
            }
```
No need to recheck if host is null three times if its not null the first time.